### DISCUSSION TITLE
Is there a plugin to manage Referral codes

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#197] [Q&A] [answered]
Hi guys,
I need a service for referrals management (e.g. "Invite a friend").
Is there, in the marketplace, a ready-to-use plugin for that?

If not, is there a service to be generalized to create a plugin for that purpose?

Thanks.

**Comment by lucascanna** [comment, marked as answer]
Hi @silversoul93,

One idea could be to use the [Temporary Password Generator](https://docs.mia-platform.eu/docs/runtime_suite/temporary-password-generator/overview).

It allows you to associate a random unique code with a payload. The functionality is offered through REST APIs.

The implementation could be something like the following:
- generate the code (POST /create) when a new user signs up and associate the code to the user
- then, the sign up API can accept a new field "referralCode" that the backend can verify (by calling the GET /verify API)
- if the verification is successful, the backend can assign the defined reward to the associate user

Let me know if this is a feasible soluton for you.
Have a nice day!
